SPahooja/SocialSports
# UI Host/Edit Match

**UI changes as follows:**
1. Added a Host tab in app bar menu, which directs users to the Location Search page
2. Added a Host Game - Location Search page
3. Added a Host Game - Details page

**build.gradle changes as follows:**
1. In top level build.gradle file:
- Added `classpath 'androidx.navigation:navigation-safe-args-gradle-plugin:2.3.5'` to pass args between fragments
2. In app level build.gradle file:
- Added `id 'androidx.navigation.safeargs.kotlin'` to pass args between fragments
- Added `id 'org.jetbrains.kotlin.android.extensions'` to use `@Parcelize`
- Set `androidExtensions {experimental = true}` to use `@Parcelize`

**New layouts:**
- `fragment_host_edit_details.xml`: create/edit game details page
- `fragment_host_location.xml`: game location search page
- `location_item.xml`: individual location card within location suggestion/search result list
- `sport_type_item.xml`: individual sport type dropdown list item

**New fragment and viewmodel for host:**
- `ui/host/location/` : game location search page
- `ui/host/details/` : create/edit game details page

**New domain files:**
- `domain/location/HostLocation.kt` : location data model for location card
- `domain/match/Match.kt` : match data model 


**Test Flow**
Run the code in an emulator/phone and see if ui is as follows:

1. Navigate to the Host tab, page should contain search bar and location suggestions list(with mock data)
<img src="https://user-images.githubusercontent.com/37742291/124406366-2a5e4080-dd0f-11eb-9d2e-81ac2b57aed3.png" width="300">

2. Search something in the search bar should return search result list (hardcoded), suggestions list should still be there below search result
<img src="https://user-images.githubusercontent.com/37742291/124406461-6e514580-dd0f-11eb-8f8b-160afd2133ec.png" width="300">

3. Select a location, user should be directed to the game details page with selected location in Location field
<img src="https://user-images.githubusercontent.com/37742291/124439633-a3798a00-dd47-11eb-9b9e-7f719d88cb7b.png" width="300">

4. Fill in the form
<img src="https://user-images.githubusercontent.com/37742291/124439704-bdb36800-dd47-11eb-8c2d-205c406142c8.png" width="300">
<img src="https://user-images.githubusercontent.com/37742291/124406539-9fca1100-dd0f-11eb-9438-d931e28fbf2d.png" width="300">
<img src="https://user-images.githubusercontent.com/37742291/124406542-a35d9800-dd0f-11eb-97cb-e51a45672d80.png" width="300">

5. Users are not allowed to submit without required fields (all fields are required except `Rules Description`)
<img src="https://user-images.githubusercontent.com/37742291/124439783-d7ed4600-dd47-11eb-9fa0-352b9a7dc113.png" width="300">

6. **EDIT EXISTING GAME** 
The form on game details page should be pre-filled with existing game information if a user is to edit a game
<img src="https://user-images.githubusercontent.com/37742291/124439562-90ff5080-dd47-11eb-83f4-5fd281172ad2.png" width="300">

For now, this can be tested by passing another argument to `HostDetailsFragment`, so add a `matchId = "id"` parameter after `locationItem` to 
`val action = HostLocationFragmentDirections.actionNavigationHostToHostEditDetails(locationItem)` in `ui/host/location/LocationListAdapter.kt` should pre-fill the form with mock data.
